tests/function-controller/testsuite/testsuite.go

```diff
@@ -70,30 +90,78 @@ func New(restConfig *rest.Config, cfg Config, t *testing.T, g *gomega.GomegaWith
 }
 
 func (t *TestSuite) Run() {
-	t.t.Log("Creating namespace...")
-	namespace, err := t.namespace.Create()
+	t.t.Logf("Creating namespace %s and default broker...", t.namespace.GetName())
+	ns, err := t.namespace.Create()
 	failOnError(t.g, err)
 
 	t.t.Log("Creating function...")
-	functionDetails := t.getFunction()
-	resourceVersion, err := t.function.Create(functionDetails)
+	functionDetails := t.getFunction(helloWorld)
+	fnResourceVersion, err := t.function.Create(functionDetails)
 	failOnError(t.g, err)
 
 	t.t.Log("Waiting for function to have ready phase...")
-	err = t.function.WaitForStatusRunning(resourceVersion)
+	err = t.function.WaitForStatusRunning(fnResourceVersion)
+	failOnError(t.g, err)
+
+	t.t.Log("Waiting for broker to have ready phase...")
```

Comment:
Should we wait first for broker, then for function?

Reply:
nah, because knative creates broker along with properly labeled namespace, so if I wait for broker immediately after namespace creation I'd have to handle the case where the broker is not yet created, and this way it's 100% sure that broker is there, but I just check its status

Reply:
it's either this order (which, in truth, doesn't matter - trigger just needs to be created after broker, even better when created after broker is ready, which I already do) or handling whole new case - for me the choice is damn obvious